plugins/Jobs.cpp

```diff
@@ -1038,10 +1039,26 @@ void BedrockJobsCommand::process(SQLite& db) {
             }
         }
 
-        string safeNewNextRun = "";
         // If this is set to repeat, get the nextRun value
+        string safeNewNextRun = "";
         if (!repeat.empty()) {
-            safeNewNextRun = _constructNextRunDATETIME(nextRun, lastRun, repeat);
+            // For all jobs, the last time at which they were scheduled is the currently stored 'nextRun' time
+            string lastScheduled = nextRun;
+
+            // Except for jobs with 'retryAfter' + 'repeat' based on `SCHEDULED`. With 'retryAfter', in GetJob we updated 'nextRun'
+            // to a failure check interval, eg 5 minutes. To account for this here when finishing the job, we subtract
+            // 'retryAfter' from 'nextRun' to get back the originally scheduled time which was 'nextRun' when the job ran.
+            if (!retryAfter.empty() && SToUpper(repeat).find("SCHEDULED") != string::npos) {
+                SQResult scheduledJobResult;
+                if (!db.read("SELECT DATETIME(nextRun, REPLACE(retryAfter, '+', '-')) "
```

Comment:
We already have the nextRun, no? You don't need to query the DB (even if we use sql to perform date aritmetics)

Reply:
Yeah, can't we just do `SELECT DATETIME(nextRun, REPLACE(retryAfter, '+', '-'))`?

But also, do we need to support the case where the `+` isn't explicit?

Reply:
>But also, do we need to support the case where the + isn't explicit?

I think it must start with `+` ? https://github.com/Expensify/Bedrock/blob/c2e74cfccb46b1d3aec01ce255dd6167c552247b/libstuff/libstuff.cpp#L2540 (or `-`, which confuses me)

Reply:
lol I doubt we are using the `-` feature and I am not even sure it is useful?

Reply:
Indeed I don't understand why you'd want a repeat/retryAfter starting with a `-`... Maybe we just remove this `-` @tylerkaraszewski ?

Reply:
`SIsValidSQLiteDateModifier` isn't just for repeats, it gets used for stuff like "the first day of next month - 1 day".